whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java

```diff
@@ -1,484 +1,499 @@
+			tableToFieldInfos = processCsvFiles(dbSettings);
+		} else
+			tableToFieldInfos = processDatabase(dbSettings);
+		generateReport(tableToFieldInfos, filename);
+	}
+
+	private Map<String, List<FieldInfo>> processDatabase(DbSettings dbSettings) {
+		Map<String, List<FieldInfo>> tableToFieldInfos = new HashMap<String, List<FieldInfo>>();
+		RichConnection connection = new RichConnection(dbSettings.server, dbSettings.domain, dbSettings.user, dbSettings.password, dbSettings.dbType);
+		connection.setVerbose(false);
+		connection.use(dbSettings.database);
+
+		dbType = dbSettings.dbType;
+		database = dbSettings.database;
+
+		for (String table : dbSettings.tables) {
+			List<FieldInfo> fieldInfos = processDatabaseTable(table, connection);
+			tableToFieldInfos.put(table, fieldInfos);
+		}
+
+		connection.close();
+		return tableToFieldInfos;
+	}
+
+	private Map<String, List<FieldInfo>> processCsvFiles(DbSettings dbSettings) {
+		delimiter = dbSettings.delimiter;
+		Map<String, List<FieldInfo>> tableToFieldInfos = new HashMap<String, List<FieldInfo>>();
+		for (String table : dbSettings.tables) {
+			List<FieldInfo> fieldInfos = processCsvFile(table);
+			String tableName = new File(table).getName();
+			if (!tableToFieldInfos.containsKey(tableName)) {
+				tableToFieldInfos.put(tableName, fieldInfos);
+			} else {
+				tableToFieldInfos.put(table, fieldInfos);
+			}
+
+		}
+		return tableToFieldInfos;
+	}
+
+	private void generateReport(Map<String, List<FieldInfo>> tableToFieldInfos, String filename) {
+		System.out.println("Generating scan report");
+		removeEmptyTables(tableToFieldInfos);
+		List<String> tables = new ArrayList<String>(tableToFieldInfos.keySet());
+		Collections.sort(tables);
+
+		SXSSFWorkbook workbook = new SXSSFWorkbook(100); // keep 100 rows in memory, exceeding rows will be flushed to disk
+
+		// Create overview sheet
+		Sheet sheet = workbook.createSheet("Overview");
+		if (!scanValues) {
+			addRow(sheet, "Table", "Field", "Type", "N rows");
+			for (String table : tables) {
+				for (FieldInfo fieldInfo : tableToFieldInfos.get(table))
+					addRow(sheet, table, fieldInfo.name, fieldInfo.getTypeDescription(), Long.valueOf(fieldInfo.rowCount));
+				addRow(sheet, "");
+			}
+		} else {
+			addRow(sheet, "Table", "Field", "Type", "Max length", "N rows", "N rows checked", "Fraction empty");
+			for (String table : tables) {
+				for (FieldInfo fieldInfo : tableToFieldInfos.get(table))
+					addRow(sheet, table, fieldInfo.name, fieldInfo.getTypeDescription(), Integer.valueOf(fieldInfo.maxLength), Long.valueOf(fieldInfo.rowCount),
+							Long.valueOf(fieldInfo.nProcessed), fieldInfo.getFractionEmpty());
+				addRow(sheet, "");
+			}
+
+			// Create per table sheets
+			for (String table : tables) {
+				String sheetName = createSheetName(table);
+
+				if (workbook.getSheet(sheetName) != null) {
+					System.out.println("Worksheet already exists in the scan report, the current table is skipped: " + sheetName);
```

Comment:
This should probably be a UI message? Suggested alternative, add a `tries` parameter to `createSheetName`:
```java
String sheetName = null;
int tries = 0;
while (sheetName == null && tries < 10) {
  sheetName = createSheetName(table, tries);
  if (workbook.getSheet(sheetName) != null) {
    sheetName = null;
  }
  tries++;
}
if (sheetName == null) {
// throw some warning
}

String createSheetName(String tableName, int tries) {
  String name = tableName;
  if (name.length() > 29) {
    name = name.substring(0, 29);
  }
  if (tries > 0) {
    name += "_" + tries;
  }
  return tries;
}
```

Reply:
Implemented a more rigorous, less elegant, approach and prepended a sheet index to all long names. This allows RiaH to get the corresponding sheet to load value counts, with backwards compatibility to older scan reports. See commit [add9f42](https://github.com/OHDSI/WhiteRabbit/pull/162/commits/add9f42481fc6b3120fbe8461bfb5d5b3a956307).

The `System.out.println` is redirected to a messages box in the UI.